src/Component/Histories/Restaurant/Restaurant.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { getExternalRestaurantUrl } from '../../../Util'
+
+// Restaurant :: Props -> React.Component
+export default ({
+  restaurant,
+  history,
+}) =>
+  <div className="history">
+    { restaurant && restaurant.image
+      ? <figure><img src={ restaurant.image } alt={ restaurant.name } /></figure>
+      : <div className="image-placeholder"></div>
+    }
+
+    <div className="content">
+      <h3>{ history.restaurantName }</h3>
+
+      { restaurant &&
+        <p>{ restaurant.address } { restaurant.zipCode } { restaurant.city }, { restaurant.country }</p>
+      }
+
+      { restaurant && restaurant.phone &&
+        <p className="phone"><a href={ 'tel:' + restaurant.phone }>{ restaurant.phone }</a></p>
+      }
+
+      <a
+        href={ getExternalRestaurantUrl(history.restaurantId) }
+        target="_blank"
+        rel="noopener noreferrer"
+        className="website"
+      >Voir la fiche</a>
+    </div>
```

Comment:
Instead of checking if the restaurant object is not valid everywhere IMHO you should just add a check at the beginning and avoid to load the component if you have an invalid restaurant object.

Reply:
I check if the restaurant informations have been fetched, because I still want to load this component, as you can see at line 16 and 27, I use the history to display the restaurant name or to have its id. Then the sheet is completed when the restaurant is fetch.